File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.rs
use std::cmp;
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut min_num = prices[0];
        let mut max_profit = 0;
        for price in prices {
            let mut profit = price - min_num;
            max_profit = cmp::max(profit, max_profit);
            min_num = cmp::min(price, min_num);
        }
        max_profit
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_num = prices[0]
        max_profit = 0

        for price in prices:
            profit = price - min_num
            max_profit = max(profit, max_profit)
            min_num = min(price, min_num)
        return max_profit